sky/clouds/service_catalog/data_fetchers/fetch_gcp.py

```diff
@@ -311,13 +311,36 @@ def _get_gpus_for_zone(zone: str) -> pd.DataFrame:
             new_gpus.append({
                 'AcceleratorName': gpu_name,
                 'AcceleratorCount': count,
-                'GpuInfo': None,
+                'GpuInfo': _gpu_info_from_name(gpu_name),
                 'Region': zone.rpartition('-')[0],
                 'AvailabilityZone': zone,
             })
     return pd.DataFrame(new_gpus).reset_index(drop=True)
 
 
+def _gpu_info_from_name(name: str) -> Optional[Dict[str, List[Dict[str, Any]]]]:
+    """Hard-codes the GPU memory info for certain GPUs.
+
+    Reference: https://cloud.google.com/compute/docs/gpus
+    """
+    name_to_gpu_memory_in_mib = {
+        'L4': 24 * 1024,
+        'A100-80GB': 80 * 1024,
+        'A100': 40 * 1024,
+        'H100': 80 * 1024,
+        'P4': 8 * 1024,
+        'T4': 16 * 1024,
+        'V100': 16 * 1024,
```

Comment:
Should we add [P100](https://cloud.google.com/compute/docs/gpus#p100-gpus) too?

Reply:
BTW, generating a catalog from this branch removed GCP P100 from my `sky show-gpus P100`:

```
# Master catalog
(base) ➜  ~ sky show-gpus P100
GPU   QTY  CLOUD  INSTANCE_TYPE       DEVICE_MEM  vCPUs  HOST_MEM  HOURLY_PRICE  HOURLY_SPOT_PRICE  REGION
P100  1    Azure  Standard_NC6s_v2    -           6      112GB     $ 2.070       $ 0.207            eastus
P100  2    Azure  Standard_NC12s_v2   -           12     224GB     $ 4.140       $ 0.414            eastus
P100  4    Azure  Standard_NC24rs_v2  -           24     448GB     $ 9.108       $ 0.911            eastus
P100  4    Azure  Standard_NC24s_v2   -           24     448GB     $ 8.280       $ 0.828            eastus
P100  1    GCP    n1-highmem-8        -           8      52GB      $ 1.933       $ 0.679            us-central1
P100  2    GCP    n1-highmem-16       -           16     104GB     $ 3.866       $ 1.357            us-central1
P100  4    GCP    n1-highmem-32       -           32     208GB     $ 7.733       $ 2.714            us-central1
P100  1    OCI    VM.GPU2.1           16GB        24     72GB      $ 1.275       -                  eu-amsterdam-1
P100  2    OCI    BM.GPU2.2           16GB        56     256GB     $ 2.550       -                  eu-amsterdam-1

# Catalog from this branch
(base) ➜  ~ sky show-gpus P100
GPU   QTY  CLOUD  INSTANCE_TYPE       DEVICE_MEM  vCPUs  HOST_MEM  HOURLY_PRICE  HOURLY_SPOT_PRICE  REGION
P100  1    Azure  Standard_NC6s_v2    -           6      112GB     $ 2.070       $ 0.207            eastus
P100  2    Azure  Standard_NC12s_v2   -           12     224GB     $ 4.140       $ 0.414            eastus
P100  4    Azure  Standard_NC24rs_v2  -           24     448GB     $ 9.108       $ 0.911            eastus
P100  4    Azure  Standard_NC24s_v2   -           24     448GB     $ 8.280       $ 0.828            eastus
P100  1    OCI    VM.GPU2.1           16GB        24     72GB      $ 1.275       -                  eu-amsterdam-1
P100  2    OCI    BM.GPU2.2           16GB        56     256GB     $ 2.550       -                  eu-amsterdam-1
```

More generally, does this change handle any new GPUs that may get added but are not updated in the `name_to_gpu_memory_in_mib` dict?

[Here's the catalog](https://gist.github.com/romilbhardwaj/49ddabe7d944b2de9487dcef78766b11) that got generated for me for reference.

Reply:
Great catch! Fixed both P100 and missing-GPU-info-not-shown issue (tested with this repro).